components/scream/src/physics/shoc/shoc_diag_second_moments_srf_impl.hpp

```diff
@@ -0,0 +1,43 @@
+
+#ifndef SHOC_DIAG_SECOND_MOMENTS_SRF_IMPL_HPP
+#define SHOC_DIAG_SECOND_MOMENTS_SRF_IMPL_HPP
+
+#include "shoc_functions.hpp" // for ETI only but harmless for GPU
+#include "physics_functions.hpp" // also for ETI not on GPUs
+
+namespace scream {
+namespace shoc {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::shoc_diag_second_moments_srf(
+    const Spack& wthl_sfc, const Spack& uw_sfc, const Spack& vw_sfc,
+    Spack& ustar2, Spack& wstar)
```

Comment:
This function is to create boundary data. Boundary data exists in only one scalar slot of a pack. Each of the inputs and outputs should be a Real, not a pack. `*_sfc` data will be prescribed as a single scalar per field per column. This function must compute the scalar ustar2 and wstar. Then, elsewhere and in a later PR, the Real values ustar2 and wstar will get injected into the slot of a pack of the column data corresponding to the boundary condition application.

Reply:
I see, good point, thanks

Reply:
@ambrad I have modified the code according to your suggestions, please review. I was not sure whether the commented out "create_mirror_view" should be there, so I didn't remove it, but I think it can be removed.

Reply:
The change to scalar looks good.